airbyte-webapp/src/pages/routes.tsx

```diff
@@ -40,7 +45,15 @@ const MainViewRoutes: React.FC<{ workspace: WorkspaceRead }> = ({ workspace }) =
     <MainView>
       <ApiErrorBoundary>
         <Routes>
-          <Route path={`${RoutePaths.Destination}/*`} element={<DestinationPage />} />
+          <Route path={RoutePaths.Destination}>
+            <Route index element={<AllDestinationsPage />} />
+            <Route path={DestinationPaths.NewDestination} element={<CreateDestinationPage />} />
+            <Route path={DestinationPaths.NewConnection} element={<CreationFormPage />} />
+            <Route path={DestinationPaths.Root} element={<DestinationItemPage />}>
+              <Route path={DestinationPaths.Settings} element={<DestinationSettingsPage />} />
+              <Route index element={<DestinationOverviewPage />} />
+            </Route>
+          </Route>
```

Comment:
This is the main improvement introduced in this PR - we can see the route structure clearly.